packages/inferno-router/src/resolveLoaders.ts

```diff
@@ -0,0 +1,198 @@
+import { isNullOrUndef, isUndefined } from "inferno-shared";
+import { matchPath } from "./matchPath";
+import type { TLoaderData, TLoaderProps } from "./Router";
+import { Switch } from "./Switch";
+
+export function resolveLoaders(loaderEntries: TLoaderEntry[]): Promise<Record<string, TLoaderData>> {
```

Comment:
IMO this could maybe be async method then it could await on Promise.all  call to make it slightly cleaner

Reply:
I tried that from the outset, but this added async/await code to the transpiled bundle and it grows substantially. So I replaced all async/await with promise/then.

ref: https://github.com/infernojs/inferno/pull/1621#issuecomment-1357294390